nntrainer/graph/graph_core.cpp

```diff
@@ -167,20 +167,22 @@ void GraphCore::ensureName(GraphNode &node, const std::string &prefix,
   node_names.insert(name);
 }
 
-void GraphCore::removeLastNode() {
-  auto last_node = Sorted.back();
-
-  Sorted.pop_back();
-
-  /// fix index after last node has been deleted.
-  auto after = node_list.erase(node_list.begin() + last_node->getIndex());
-  auto after_index = last_node->getIndex();
+void GraphCore::replaceNode(std::shared_ptr<GraphNode> from,
+                            std::shared_ptr<GraphNode> to) {
+  unsigned int idx = from->getIndex();
+  to->setIndex(idx);
+  node_list[idx] = to;
+}
 
-  for (; after != node_list.end(); ++after) {
-    (*after)->setIndex(after_index++);
+void GraphCore::realizeInputOutputNode() {
+  for (auto iter = cbegin(); iter != cend(); ++iter) {
+    if (iter->getInputConnections()[0] == "__data__") {
```

Comment:
Please note that this will have to change after merging v2. it will compile so it will be hard to find the problem.

Reply:
yes, this has been added to the list of manual merging along with padding :)